experimental/devices/mfrc522/mfrc522.go

```diff
@@ -0,0 +1,767 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package mfrc522 controls a Mifare RFID card reader.
+//
+// The code is largely ported from Python library : https://github.com/mxgxw/MFRC522-python
+//
+// The datasheet is available at : https://www.nxp.com/docs/en/data-sheet/MFRC522.pdf
+package mfrc522
+
+import (
+	"errors"
+	"fmt"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+	"time"
+)
+
+// Dev is an handle to an MFRC522 RFID reader.
+type Dev struct {
+	resetPin      gpio.PinOut
+	irqPin        gpio.PinIn
+	Authenticated bool
+	maxSpeedHz    int64
+	spiDev        spi.Conn
+}
+
+// DefaultKey provides the default bytes for card authentication for method B.
+var DefaultKey = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+
+const maxSpeed = int64(1000000)
+
+/*
+NewSPI creates and initializes the RFID card reader attached to SPI.
+	spiPort - the SPI device to use.
+	resetPin - reset GPIO pin.
+	irqPin - irq GPIO pin.
+*/
+func NewSPI(spiPort spi.Port, resetPin, irqPin string) (*Dev, error) {
+
+	spiDev, err := spiPort.Connect(maxSpeed, spi.Mode0, 8)
+	if err != nil {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	dev := &Dev{
+		spiDev:     spiDev,
+		maxSpeedHz: maxSpeed,
+	}
+
+	pin := gpioreg.ByName(resetPin)
+	dev.resetPin = pin
+	dev.resetPin.Out(gpio.High)
+
+	pin = gpioreg.ByName(irqPin)
+	dev.irqPin = pin
+	dev.irqPin.In(gpio.PullUp, gpio.FallingEdge)
+
+	err = dev.Init()
+
+	return dev, nil
+}
+
+var initCommands = [][]byte{
+	{commands.TModeReg, 0x8D},
+	{commands.TPrescalerReg, 0x3E},
+	{commands.TReloadRegL, 30},
+	{commands.TReloadRegH, 0},
+	{commands.TxAutoReg, 0x40},
+	{commands.ModeReg, 0x3D},
+}
+
+// Init initializes the RFID chip.
+func (r *Dev) Init() error {
+	err := r.Reset()
+	if err != nil {
+		return err
+	}
+	for _, cmdData := range initCommands {
+		err = r.devWrite(int(cmdData[0]), cmdData[1])
+		if err != nil {
+			return err
+		}
+	}
+	err = r.SetAntenna(true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Dev) devWrite(address int, data byte) (err error) {
```

Comment:
Do not use named return values unless you need to modify the return value in a defer function. Here and below.

In this case, the whole function reduces to one or two lines.